File: Bubble Sort/Bubble Sort.pl
# BUBBLE SORT IN PERL

# function algorithm that returns the sorted array
# algoritmo da funcao que retorna o array ordenado
sub bubbleSort
{
	@arr = @_;

	for ($i = 0; $i < scalar @arr - 1; $i++)
	{
		for ($j = $i + 1; $j < scalar @arr; $j++)
		{
			if ($arr[$i] > $arr[$j])
			{
				$swap = $arr[$i];
				$arr[$i] = $arr[$j];
				$arr[$j] = $swap;
			}
		}
	}

	return @arr;
}

# code only to call and test the sort function
# código apenas para chamar e testar a função de ordenação
for ($i = 0; $i < 10; $i++)
{
	$arr[$i] = <>;
}

@arrSorted = bubbleSort @arr;

print "\nArray Sorted:\n";

foreach $a (@arrSorted)
{
	print $a;
}

File: Bubble Sort/Bubble Sort.lua
-- BUBBLE SORT IN LUA

-- function algorithm that returns the sorted array
-- algoritmo da função que retorna o array ordenado
function bubbleSort(arr)
	for i = 1, #arr - 1 do
		for j = i + 1, #arr do
			if (arr[i] > arr[j]) then
				arr[i], arr[j] = arr[j], arr[i];
			end
		end
	end

	return arr;
end

-- code only to call and test the sort function
-- código apenas para chamar e testar a função de ordenação
arr = {5, 2, -3, 10, 23, 99, -1, 7, 93, 0};
arrSorted = bubbleSort(arr)

print("Array Sorted")

for i = 1, #arrSorted do
	print(arrSorted[i]);
end